utils/random_payload_generator.py

```diff
@@ -0,0 +1,108 @@
+# Copyright © 2019, 2020 Pavel Tisnovsky
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generator of random payload for testing REST API, message consumers, test frameworks etc."""
+
+import string
+import random
+
+
+class RandomPayloadGenerator:
+    """Generator of random payload for testing API."""
+
+    def __init__(self):
+        """Initialize the random payload generator."""
+        self.iteration_deep = 0
+        self.max_iteration_deep = 2
+        self.max_dict_key_length = 10
+        self.max_string_length = 20
+        self.dict_key_characters = string.ascii_lowercase + string.ascii_uppercase + "_"
+        self.string_value_characters = (string.ascii_lowercase + string.ascii_uppercase +
+                                        "_" + string.punctuation + " ")
+
+    def generate_random_string(self, n, uppercase=False, punctuations=False):
+        """Generate random string of length=n."""
+        prefix = random.choice(string.ascii_lowercase)
+        mix = string.ascii_lowercase + string.digits
+
+        if uppercase:
+            mix += string.ascii_uppercase
+        if punctuations:
+            mix += string.punctuation
+
+        suffix = ''.join(random.choice(mix) for _ in range(n - 1))
+        return prefix + suffix
+
+    def generate_random_key_for_dict(self, data):
+        """Generate a string key to be used in dictionary."""
+        existing_keys = data.keys()
+        while True:
+            new_key = self.generate_random_string(10)
+            if new_key not in existing_keys:
+                return new_key
+
+    def generate_random_list(self, n):
+        """Generate list filled in with random values."""
+        return [self.generate_random_payload((int, str, float, bool, list, dict)) for i in range(n)]
+
+    def generate_random_dict(self, n):
+        """Generate dictionary filled in with random values."""
+        dict_content = (int, str, list, dict)
+        return {self.generate_random_string(10): self.generate_random_payload(dict_content)
+                for i in range(n)}
+
+    def generate_random_list_or_string(self):
+        """Generate list filled in with random strings."""
+        if self.iteration_deep < self.max_iteration_deep:
+            self.iteration_deep += 1
+            value = self.generate_random_list(5)
+            self.iteration_deep -= 1
+        else:
+            value = self.generate_random_value(str)
+        return value
+
```

Comment:
I don't understand this piece of code. `self.max_iteration_deep` is always 2, and `self.iteration_deep` starts as 0...
Taking into account that, when you enter in the `if` branch you will increment and then decrement the value of `self.iteration_deep`, I don't see that it is possible to enter in the `else` branch at all.

So, `generate_random_list_or_string` always will generate a random list with 5 as length.

Reply:
the script is more universal and all attrbiutes can be changed in future, if needed